spec/features/visitors/sign_up_spec.rb

```diff
@@ -58,20 +58,22 @@
     before do
       sign_up_with_and_set_password_for('test@example.com')
       fill_in 'Mobile', with: '555-555-5555'
+      expect(Users::PhoneConfirmationController).
+        to receive(:generate_confirmation_code).and_return('1234')
```

Comment:
Although I think they are interchangeable, I prefer to use `allow` when stubbing a value to distinguish between stubs and expectations.